nexus/db-queries/src/db/datastore/trust_quorum.rs

```diff
@@ -0,0 +1,1629 @@
+            .execute_async(conn)
+            .await?;
+        Ok(())
+    }
+
+    async fn update_tq_member_share_digest_in_txn(
+        opctx: &OpContext,
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+        member: DbTrustQuorumMember,
+    ) -> Result<(), TransactionError<Error>> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+        use nexus_db_schema::schema::trust_quorum_member::dsl;
+
+        diesel::update(dsl::trust_quorum_member)
+            .filter(dsl::rack_id.eq(member.rack_id))
+            .filter(dsl::epoch.eq(member.epoch))
+            .filter(dsl::hw_baseboard_id.eq(member.hw_baseboard_id))
+            .filter(dsl::share_digest.is_null())
+            .filter(dsl::state.eq(DbTrustQuorumMemberState::Unacked))
+            .set(dsl::share_digest.eq(member.share_digest))
+            .execute_async(conn)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn update_tq_member_state_prepared_in_txn(
+        opctx: &OpContext,
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+        member: DbTrustQuorumMember,
+    ) -> Result<(), TransactionError<Error>> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+        use nexus_db_schema::schema::trust_quorum_member::dsl;
+
+        diesel::update(dsl::trust_quorum_member)
+            .filter(dsl::rack_id.eq(member.rack_id))
+            .filter(dsl::epoch.eq(member.epoch))
+            .filter(dsl::hw_baseboard_id.eq(member.hw_baseboard_id))
+            .filter(dsl::share_digest.is_not_null())
+            .filter(dsl::state.eq(DbTrustQuorumMemberState::Unacked))
+            .set(dsl::state.eq(DbTrustQuorumMemberState::Prepared))
+            .execute_async(conn)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn update_tq_encrypted_rack_secrets_in_txn(
+        opctx: &OpContext,
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+        rack_id: DbTypedUuid<RackKind>,
+        epoch: i64,
+        encrypted_rack_secrets: EncryptedRackSecrets,
+    ) -> Result<(), TransactionError<Error>> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+        let salt = Some(hex::encode(encrypted_rack_secrets.salt.0));
+        let secrets: Option<Vec<u8>> = Some(encrypted_rack_secrets.data.into());
+
+        use nexus_db_schema::schema::trust_quorum_configuration::dsl;
+
+        diesel::update(dsl::trust_quorum_configuration)
+            .filter(dsl::rack_id.eq(rack_id))
+            .filter(dsl::epoch.eq(epoch))
+            .filter(dsl::encrypted_rack_secrets_salt.is_null())
+            .filter(dsl::encrypted_rack_secrets.is_null())
+            .set((
+                dsl::encrypted_rack_secrets_salt.eq(salt),
+                dsl::encrypted_rack_secrets.eq(secrets),
+            ))
+            .execute_async(conn)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn update_tq_encrypted_rack_secrets_and_commit_in_txn(
```

Comment:
Should this be renamed slightly for the new `Committing` name? `*_and_commit` sounds like it would set it to `Committed`.